src/Stratis.Features.SQLiteWalletRepository/Tables/TempTable.cs

```diff
@@ -58,8 +87,8 @@ internal IEnumerable<string> CreateScript()
 
             var props = GetProperties(this.RowType);
 
-            if (this.Count > 0)
-                yield return $"INSERT INTO temp.{this.RowType.Name} {ObjectColumns()} VALUES {string.Join(Environment.NewLine + ",", this.Select(obj => ObjectRow(props, obj)))};";
+            foreach (IEnumerable<TempRow> batch in ((IEnumerable<TempRow>)this).Batch(500))
```

Comment:
no magic numbers pls, use constants (maybe you could have access to SQLITE_MAX_COMPOUND_SELECT)
and we may even use that const to change the value used by SQLite
https://www.sqlite.org/limits.html point n7